Package

rminnich opened the pull request.

rminnich committed:
Add a HostName to Cmd struct

The HostName is either the Host, or the HostName as found
in an .ssh/config file. Use HostName for the Dial, not Host.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
Add a SetPort function for the Cmd

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
Change port type to string

We should not determine the rules of a valid port type in
this package; that's up to the net package.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
In Dial, use config to find an IdentityFile if needed.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
Add namespace setup code to the package

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
cpu: begin implementing package-based command

This will, gradually, become equivalent to, and replace, the
old one that did not use the package.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
cpu pkg: do not start a 9p server if cmd.NameSpace is "none"

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
cpu: pkg support, pass args

This works to this point:
./cpu -old=false  v mount

2022/03/09 14:57:14 c.Run: (<nil>, "rootfs on / type rootfs (rw)\nproc on /proc type proc (rw,relatime)\nsysfs on /sys type sysfs (rw,relatime)\nfusectl on /sys/fs/fuse/connections
type fusectl (rw,relatime)\ndevpts on /dev/pts type devpts (rw,relatime,mode=600,ptmxmode=000)\ntmpfs on /dev/shm type tmpfs (rw,relatime)\ncpu on /tmp type tmpfs
(rw,relatime)\n127.0.0.1 on /tmp/cpu type 9p (rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\nrootfs on /tmp/local type rootfs
(rw)\n127.0.0.1 on /lib type 9p (rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n127.0.0.1 on /lib64 type 9p
(rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n127.0.0.1 on /usr type 9p
(rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n127.0.0.1 on /bin type 9p
(rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n127.0.0.1 on /etc type 9p
(rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n127.0.0.1 on /home type 9p
(rw,nosuid,nodev,relatime,sync,dirsync,uname=rminnich,access=client,msize=1048576,trans=fd,rfd=9,wfd=9)\n", "")

Note that the namespace worked fine, using the package code.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

orangecms commented:
What does pkg/package mean here? I am not seeing it. 🤔 

Is that about the `"github.com/u-root/cpu/cpu"` import?

rminnich commented:
> What does pkg/package mean here? I am not seeing it. 🤔
> 
> Is that about the `"github.com/u-root/cpu/cpu"` import?

The culture has changed. pkg is no longer the standard. Packages start at top level now.
https://github.com/golang-standards/project-layout/issues/10

and you're supposed to use internal/ where possible to hide packages you don't want people to see

for $MODULES$

but cpu/cpu is still a package :-)

rminnich commented:
anyway it's not ready, this will be a series, *almost* there but not quite

rminnich committed:
package support: get cpu working when namespace="none"

There was an interesting gotcha with this line:
       c.Env = append(c.Env, "CPUNONCE="+c.nonce.String())

when the nonce was not set, it seemed to create a corrupt environment,
and the cmd.Start would return EINVAL.

Kernels that use integer error return should not be allowed to exist.
But they do, sadly.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
cpu package: set up interactive support functions

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
cpu: cleanup: remove unused cmd() function

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
checkpoint commit -- ptys are hateful.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

rminnich committed:
rename cpu package to client

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>
